test/e2e/performanceprofile/functests/utils/cgroup/v1/v1.go

```diff
@@ -0,0 +1,38 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/cgroup/config"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/pods"
+)
+
+type Manager struct {
+}
+
+func (m *Manager) GetConfig(ctx context.Context, c *kubernetes.Clientset, pod *corev1.Pod, containerName string) (*config.Config, error) {
```

Comment:
single function interface suggests you may want to return... a function rather than an object. Unless there's a very clear path about adding more methods

Reply:
Yes we do.
It should have additional interfaces to retrieve pod config and container's child config (we have childs in crun and mixedCpus)

Reply:
fair enough, so using a type (vs a function) has its merits

Reply:
Now I noticed I wrote additional interfaces. I meant I want to add additional functions to the same interface, where each manager implement it own logic

Reply:
NP, somehow parsed the statement as intended rather than as written :)